packages/go_router/lib/src/parser.dart

```diff
@@ -216,6 +216,9 @@ class GoRouteInformationParser extends RouteInformationParser<RouteMatchList> {
         return baseRouteMatchList!.uri.toString() != newMatchList.uri.toString()
             ? newMatchList
             : baseRouteMatchList;
+      case NavigatingType.pop:
+        // Will do nothing.
```

Comment:
This needs some more description. maybe soemthing like
```
RouterDelegate is updating its RouteMatchList. Ignore this update to avoid overriding the RouterDelegate's RouteMatchList
```